fix: Update Google Cloud Enterprise Search to Vertex AI Search

holtskinner opened the pull request:
- Description: Google Cloud Enterprise Search was renamed to Vertex AI Search
  - https://cloud.google.com/blog/products/ai-machine-learning/vertex-ai-search-and-conversation-is-now-generally-available
  - This PR updates the documentation and Retriever class to use the new terminology.
  - Changed retriever class from `GoogleCloudEnterpriseSearchRetriever` to `GoogleVertexAISearchRetriever`
  - Updated documentation to specify that `extractive_segments` requires the new [Enterprise edition](https://cloud.google.com/generative-ai-app-builder/docs/about-advanced-features#enterprise-features) to be enabled.
  - Fixed spelling errors in documentation.
  - Change parameter for Retriever from `search_engine_id` to `data_store_id`
    - When this retriever was originally implemented, there was no distinction between a data store and search engine, but now these have been split.
  - Fixed an issue blocking some users where the api_endpoint can't be set

holtskinner committed:
fix: Update Enterprise Search to Vertex AI Search

- Product was renamed
- Future commits will update file names/retriever name and add redirects

holtskinner committed:
fix: Rename Documentation/Notebook and add redirects

holtskinner committed:
fix: Rename `GoogleCloudEnterpriseSearchRetriever` to `GoogleCloudVertexAISearchRetriever`

- Added in mapping of old class for backwards compatibility

holtskinner committed:
docs: Fix typo in link

holtskinner committed:
fix lint error - Split long line

holtskinner commented:
NOTE: I wasn't able to verify that the documentation redirect from 

docs/integrations/retrievers/google_cloud_enterprise_search

to

docs/integrations/retrievers/google_cloud_vertex_ai_search

works in vercel. Is there anything I'm missing, or do redirects not work in the preview environment?

holtskinner committed:
Change parameter for Retriever from `search_engine_id` to `data_store_id`

- When this retriever was originally implemented, there was no distinction between a data store and search engine, but now these have been split.
- Currently `search_engine_id` isn't used for anything in the Vertex AI Search API, but separating these out now in case this behavior changes.

holtskinner commented on docs/extras/integrations/retrievers/google_cloud_vertex_ai_search.ipynb:
```diff
@@ -4,18 +4,17 @@
    "metadata": {},
    "source": [
-    "# Google Cloud Enterprise Search\n",
+    "# Google Cloud Vertex AI Search\n",
```
Yes, it would be less verbose, but I want this to be clear that it's part of Google Cloud.
There's already been lots of confusion from users about the differences between [MakerSuite](https://developers.generativeai.google/products/makersuite) and all of the non-cloud Generative AI Developer Tools.
I can change this if you think it would be an improvement

holtskinner replied:
I'll change this to match the other Vertex AI integrations https://python.langchain.com/docs/integrations/llms/google_vertex_ai_palm

holtskinner committed:
fix: Lint errors in long strings

holtskinner committed:
Fix warning strings

holtskinner committed:
fix: Addressed Code Review comments

- Changed references to be Google Vertex AI Search instead of Google Cloud Vertex AI Search

leo-gan reviewed (approved):
Thanks a lot @holtskinner !
LGTM

holtskinner committed:
Merge branch 'master' into master

holtskinner commented:
Thanks! Can you or someone with access merge this in?

holtskinner committed:
Merge branch 'master' into master

leo-gan commented:
@baskaryan FYI

holtskinner committed:
Merge branch 'master' into master

baskaryan commented on libs/langchain/langchain/retrievers/google_vertex_ai_search.py:
```diff
@@ -273,3 +305,20 @@ def _get_relevant_documents(
         return documents
+
+
+class GoogleCloudEnterpriseSearchRetriever(GoogleVertexAISearchRetriever):
```
can we keep this in old file for backwards compatibility

holtskinner replied:
Sorry, missed this, I'll update it in another commit.

baskaryan replied:
not seeing it, maybe im missing something?

holtskinner replied:
I added it, then I removed it because it was confusing early users.

baskaryan replied:
this is a breaking change if we don't include that, would prefer to avoid breaks if easy

holtskinner committed:
fix: Move back `GoogleCloudEnterpriseSearchRetriever` into original file for backwards compatibility

holtskinner committed:
fix: Add API Endpoint at client creation

- Context: https://cloud.google.com/generative-ai-app-builder/docs/locations#specify_a_multi-region_for_your_data_store

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
fix: Add lint ignore for long line

holtskinner committed:
Remove extra import of `ClientOptions`

holtskinner commented:
I also added a fix for internal ticket 300400448 where datastores not in the global region don't work with the retriever.

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Improve documentation readability

holtskinner committed:
Merge branch 'master' of https://github.com/holtskinner/langchain

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Update deprecation message text

holtskinner committed:
Fix error checking for DATA_STORE_ID and SEARCH_ENGINE_ID

holtskinner committed:
Remove backwards compatibility
- It was confusing users trying to use the new version
- Anyone using the old version can stay on an old version of langchain
- Greatly simplifies code

holtskinner committed:
Merge branch 'langchain-ai:master' into master

holtskinner commented:
@baskaryan the sooner this can be merged in, the better. This is blocking several users of this Retriever who can't use it in specific data store locations.

leo-gan commented:
@baskaryan Can you, please, help with the Vercel conflict? Thanks

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner commented:
Vercel issue seems to be resolved. I think this was because of a force push I made.

leo-gan commented:
@baskaryan PR is ready

holtskinner committed:
Merge branch 'master' into master

leo-gan commented:
@holtskinner Hi , could you, please, resolve the merging issues? After that ping me and I push this PR for the review. Thanks!

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

leo-gan commented:
@baskaryan PR is ready. Thanks!

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner commented:
@baskaryan Would it be possible to get this merged in soon? Thanks

holtskinner committed:
Merge branch 'master' into master

leo-gan commented:
@baskaryan Could you, please, review it?

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner commented:
@baskaryan Hi, would you please be able to re-review and merge this PR?

leo-gan commented:
@holtskinner Hi , could you, please, resolve the merging issues? After that ping me and I push this PR for the review. Thanks!

holtskinner committed:
Merge branch 'master' into master

holtskinner commented:
Merge conflict is resolved.

holtskinner committed:
Merge branch 'master' into master

holtskinner commented:
@baskaryan @leo-gan Would either of you be able to please expedite merging this Pull Request? This is preventing quite a few Google Cloud customers from using this retriever.

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner commented:
@nfcampos @eyurtsev @agola11 @hwchase17 @hinthornw @rlancemartin If any of you could please merge this PR in soon, that would be very helpful, thank you!

holtskinner committed:
Merge branch 'langchain-ai:master' into master

hwchase17 reviewed (commented):
this is very backwards incompatible. it will break any client using it. did we consider making the changes in a more backwards compatible way?

hwchase17 commented:
if you dont want to make it backwards compatible, i would suggest just making this a net NEW thing and deprecating the old one. i can also help with that if that sounds good

holtskinner committed:
Merge branch 'master' into master

holtskinner commented:
Yes, I did originally. You can see in the commit history. However, this was confusing users that were trying to use my forked version specifically because of the parameter name changes. (Search Engines and Data Stores used to share the same ID in previous versions of the product, but now they don't, and it was too confusing to have both parameters be available).

Anyone that doesn't want to change to the new version can stay on an older version of LangChain. (It's a very straightforward change to make, I can even add a more explicit migration guide in the docs, but I already have sections that discuss the previous versions)

holtskinner committed:
docs: Add migration guide for `GoogleCloudEnterpriseSearchRetriever`

holtskinner committed:
Added back in backwards compatibility with `GoogleCloudEnterpriseSearchRetriever`

- Deprecation warning is thrown when using old class or `search_engine_id` parameter

holtskinner committed:
Fix lint errors

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Fix test lint error

holtskinner committed:
Add Vertex AI Search retriever to Google index docs

holtskinner committed:
Update doc heading

holtskinner committed:
Merge branch 'master' into master

holtskinner commented:
@hwchase17 how does this look to you?

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner commented:
Can this PR please be merged/re-reviewed soon?

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'master' into master

holtskinner commented:
@hwchase17 @baskaryan is there anything needed in order to merge this change in? It's been sitting for almost a month, and it's been impacting a lot of Google Cloud's users.

holtskinner committed:
Merge branch 'master' into master

holtskinner committed:
Merge branch 'langchain-ai:master' into master

holtskinner committed:
Merge branch 'master' into master

baskaryan commented:
sorry for delay @holtskinner, looks great now. thanks for updating!

holtskinner commented:
Thank you very much @baskaryan ! Quick Question, I noticed that this link: https://python.langchain.com/docs/integrations/retrievers/google_cloud_enterprise_search

didn't redirect to this link https://python.langchain.com/docs/integrations/retrievers/google_vertex_ai_search as I expected by setting it in [vercel.json](https://github.com/langchain-ai/langchain/pull/10513/files#diff-8c977237e5304a44de7ca45c503ad76d945e048ae39fc64f3c25529b08464c24) Is there any reason for this? Did I do it incorrectly?
